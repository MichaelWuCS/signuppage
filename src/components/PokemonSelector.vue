<template>
  <div>
    <input
      id="inputbox"
      v-model="pokemonId"
      type="text"
      placeholder="Enter a Pokémon's name or pokédex number"
    />
    <br />
    <button type="button" @click="setPokemonId">Input!</button>
  </div>
</template>

<script>
export default {
  name: "PokemonSelector",
  data() {
    return {
      pokemonId: ""
    };
  },
  methods: {
    setPokemonId() {
      this.$store.dispatch("setPokemonIdAsync", this.pokemonId);
      this.$router.push({
        name: "pokemon",
        params: { pokemonId: this.pokemonId }
      });
    }
  }
};
</script>

<style scoped>
#inputbox {
  margin-bottom: 0px;
  max-width: 300px;
  display: block;
  width: 100%;
  padding: 10px;
  border: 1px solid #dddddd;
  border-radius: 5px;
}
</style>