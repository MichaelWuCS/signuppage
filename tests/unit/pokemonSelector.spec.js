import PokemonSelector from '@/components/PokemonSelector.vue';
import { shallowMount } from '@vue/test-utils';
import { config } from '@vue/test-utils';

describe('pokemonSelector', () => {
  test('pokemonSelector should not take in any props', () => {
    //Arrange
    const parameter = 'Hello';

    //Act
    const wrapper = shallowMount(PokemonSelector, {});
    const result = wrapper.html();

    //Assert
    expect(result).toMatchSnapshot();
  });
});
