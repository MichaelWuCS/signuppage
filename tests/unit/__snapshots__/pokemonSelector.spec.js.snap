// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pokemonSelector pokemonSelector should not take in any props 1`] = `<div><input id="inputbox" type="text" placeholder="Enter a Pokémon's name or pokédex number"> <br> <button type="button">Input!</button></div>`;
